### ISSUE TITLE
How to use AddTypeActivatedCheck for HealthChecks. I get error.

### ISSUE THREAD

**Issue by Eilon** [aspnet/Extensions#684]
_From @DenKn on Wednesday, 12 December 2018 11:07:07_

# General
 Microsoft.NETCore.App 2.2.0
 Windows 10

I have class.

```c#
public class CustomHealthCheck: IHealthCheck
    {
        private readonly string info;
        
        public CustomHealthCheck(string info)
        {
            this.info= info;
        }
    
        public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = new CancellationToken())
        {                       
                return await Task.FromResult(HealthCheckResult.Healthy($"{info}"));
        }
    }
```

If in my service I use such code
```c#
public virtual void ConfigureServices(IServiceCollection services)
        {
            services.AddMvcCore()
                .AddControllersAsServices();            

            services.AddHealthChecks().AddTypeActivatedCheck<CustomHealthCheck>("test",  "hi");
        }

public virtual void Configure(IApplicationBuilder app, IApplicationLifetime appLifetime)
{
           var options = new HealthCheckOptions();
            options.ResultStatusCodes[HealthStatus.Unhealthy] = StatusCodes.Status503ServiceUnavailable;
            options.ResponseWriter = async (ctx, rpt) =>
            {
                var result = new FullHealthResult(rpt.Status,
                    rpt.Entries.Select(e => new HealthCheckResult(e.Value.Status, e.Value.Description, e.Value.Exception)).ToArray());              
                ctx.Response.ContentType = MediaTypeNames.Application.Json;
                await ctx.Response.WriteAsync(result.ToJson());
            };
            app.UseHealthChecks("/health", options);
}
```
and call _http://localhost/health_ I get **Internal server error**.

But if I use the code below and call _http://localhost/health_  it works correctly.
```c#
public virtual void ConfigureServices(IServiceCollection services)
        {
            services.AddMvcCore()
                .AddControllersAsServices();            

            services.AddHealthChecks().Add(new HealthCheckRegistration("test",
                s => new CustomHealthCheck("hi"), null, null))
        }
```
What's problem?

_Copied from original issue: dotnet/core#2138_

**Comment by rynowak** [comment]
Hi @denkn - it looks like you're running into this issue: https://github.com/aspnet/Extensions/issues/598

**Comment by rynowak** [comment]
This is fixed in 3.0-preview4
